cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlClientTest.java

```diff
@@ -390,4 +407,124 @@ private void ccBrokerDoesNotExist(Vertx vertx, VertxTestContext context, Abstrac
                 throw new IllegalArgumentException("The " + endpoint + " endpoint is invalid for this test");
         }
     }
+
+    @Test
+    public void testCCUserTaskNoDelay(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+        runTest(vertx, context, MockCruiseControl.REBALANCE_NO_GOALS_RESPONSE_UTID, 0);
+    }
+
+    @Test
+    public void testCCUserTaskNoDelayVerbose(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+        runTest(vertx, context, MockCruiseControl.REBALANCE_NO_GOALS_VERBOSE_RESPONSE_UTID, 0);
+    }
+
+    @Test
+    public void testCCUserTaskDelay(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+        runTest(vertx, context, MockCruiseControl.REBALANCE_NO_GOALS_RESPONSE_UTID, 3);
+    }
+
+    @Test
+    public void testCCUserTaskDelayVerbose(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+        runTest(vertx, context, MockCruiseControl.REBALANCE_NO_GOALS_VERBOSE_RESPONSE_UTID, 3);
+    }
+
+    @Test
+    public void testMockCCServerPendingCallsOverride(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+        CruiseControlApi client = cruiseControlClientProvider(vertx);
+        String userTaskID = MockCruiseControl.REBALANCE_NO_GOALS_RESPONSE_UTID;
+
+        int pendingCalls1 = 2;
+        Checkpoint firstPending = context.checkpoint(pendingCalls1);
+        int pendingCalls2 = 4;
+        Checkpoint secondPending = context.checkpoint(pendingCalls2);
```

Comment:
AFAIU this test runs the same thing but twice ... first time with 2 pending calls and second time with 4 pending calls? Am I right? If yes, why?

Reply:
TBH, I'm just doing refactoring here, not changing any test. This tests was simply moved from MockCruiseControlTest as it was actually testing the CC client.

Reply:
Interesting ... Git is blaming me ... who knows why I wrote the test this way. I would leave this out for this PR, I will check the code deeply and if no reasons I will just remove one of them.